packages/odyssey-react-mui/src/labs/GroupPicker.tsx

```diff
@@ -0,0 +1,247 @@
+import { HTMLAttributes, memo, useCallback } from "react";
+
+import { AutocompleteProps } from "../Autocomplete";
+import { Field } from "../Field";
+import { Subordinate } from "../Typography";
+import { useOdysseyDesignTokens } from "../OdysseyDesignTokensContext";
+
+import { UserIcon, GridIcon, GroupIcon } from "../icons.generated";
+
+export type GroupPickerOptionType = {
+  id: string;
+  name: string;
+  description: string;
+  logo?: string;
+  usersCount?: number;
+  appsCount?: number;
+  groupPushMappingsCount?: number;
+};
+
+export type GroupPickerProps<
+  GroupPickerOptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined
+> = {
+  /**
+   * Hide selected options
+   */
+  filterSelectedOptions?: MuiAutocompleteProps<
+    GroupPickerOptionType,
+    HasMultipleChoices,
+    undefined,
+    IsCustomValueAllowed
+  >["filterSelectedOptions"];
+} & AutocompleteProps<
+  GroupPickerOptionType,
+  HasMultipleChoices,
+  IsCustomValueAllowed
+>;
+
+const GroupPicker = <
+  optionType extends GroupPickerOptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined
+>({
+  hasMultipleChoices,
+  isCustomValueAllowed,
+  isDisabled,
+  isLoading,
+  isOptional = false,
+  isReadOnly,
+  hint,
+  label,
+  onChange,
+  onInputChange,
+  options,
+  value,
+  filterSelectedOptions,
+  testId,
+}: GroupPickerProps<optionType, HasMultipleChoices, IsCustomValueAllowed>) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const isOptionEqualToValue = useCallback(
+    (sourceValue: optionType, targetValue: optionType) => {
+      return sourceValue.id === targetValue.id;
+    },
+    []
+  );
+
+  const getOptionLabel = useCallback((option: string | optionType) => {
+    return typeof option === "string" ? option : option.name;
+  }, []);
+
+  const renderOption = useCallback(
+    (props: HTMLAttributes<HTMLLIElement>, option: optionType) => {
+      return (
+        <li {...props} key={option.id}>
+          <Box
+            sx={{
+              display: "flex",
+              flexDirection: "row",
+              alignItems: "top",
+            }}
+          >
+            <Box sx={{ paddingRight: odysseyDesignTokens.Spacing2 }}>
+              <MuiAvatar
+                sx={{
+                  backgroundColor: "transparent",
+                  width: 24,
+                  height: 24,
+                }}
+                src={option.logo}
+              >
+                &nbsp;
+              </MuiAvatar>
+            </Box>
+            <Box>
+              {option.name}
+              <Subordinate>{option.description}</Subordinate>
+              <Box
+                sx={{
+                  display: "flex",
+                  flexDirection: "row",
+                  paddingTop: odysseyDesignTokens.Spacing1,
+                }}
+              >
+                {option.usersCount !== undefined && (
```

Comment:
Can we swap this to `typeof` because I like positive checks rather than negative ones. Also, this would've allowed `null` the way it is today.

```suggestion
                {typeof option.usersCount === 'number' && (
```

Reply:
We have 3 of these in the code.